src/test/java/com/amazon/ion/CloneTest.java

```diff
@@ -266,30 +269,151 @@ public void modifyAfterCloneDoesNotChangeOriginal() {
+        IonStruct struct = ionSystem.newEmptyStruct();
+        struct.put("a", ionSystem.newSymbol("foo"));
+        struct.put("b", ionSystem.newSymbol("foo"));
+        struct.put("c", ionSystem.newSymbol("foo"));
+        struct.put("number", ionSystem.newDecimal(1e-2));
+        IonStruct child1 = ionSystem.newEmptyStruct();
+        for (int i = 0; i < 8; i++) {
+            child1.put("field" + i, attribute.clone());
+        }
+        struct.put("child1", child1);
+        IonStruct child2 = ionSystem.newEmptyStruct();
+        for (int i = 0; i < 1000; i++) {
+            child2.put("beforeTarget" + i, attribute.clone());
+        }
+        child2.put("target", attribute);
+        for (int i = 0; i < 400; i++) {
+            child2.put("afterTarget" + i, attribute.clone());
+        }
+        struct.put("child2", child2);
+        return struct;
+    }
+
+    /**
+     * Verifies that a read-only struct's large nested struct can be accessed via multiple threads concurrently.
+     * @param cloneStruct true if the struct should be cloned before it is made read-only.
+     * @param makeParentReadOnly if true, the top-level struct will be made only (cascading to all children); if false,
+     *                           only the child struct to be accessed concurrently will be made read-only.
+     */
+    public void testReadOnlyIonStructMultithreadedNestedAccess(boolean cloneStruct, boolean makeParentReadOnly) {
+        IonStruct struct;
+        if (cloneStruct) {
+            struct = createIonStructWithLargeChildStructs().clone();
+        } else {
+            struct = createIonStructWithLargeChildStructs();
+        }
+        if (makeParentReadOnly) {
+            struct.makeReadOnly();
+        } else {
+            struct.get("child2").makeReadOnly();
+        }
+
+        int numberOfTasks = 100;
+        int numberOfIterationsPerTask = 5;
+        AtomicInteger success = new AtomicInteger();
+        AtomicInteger error = new AtomicInteger();
+        List<CompletableFuture<Void>> tasks = new ArrayList<>();
+
+        for (int task = 0; task < numberOfTasks; task++) {
+            tasks.add(CompletableFuture.runAsync(
+                () -> {
+                    for (int i = 0; i < numberOfIterationsPerTask; i++) {
+                        IonStruct child2 = (IonStruct) struct.get("child2");
+                        String value = child2 == null || child2.get("target") == null
+                            ? null
+                            : ((IonStruct) child2.get("target")).get("a").toString();
+
+                        if (value == null) {
+                            error.getAndIncrement();
+                            break;
+                        }
+                        success.getAndIncrement();
+                    }
+                }
+            ));
+        }
+        tasks.forEach(CompletableFuture::join);
+
+        assertTrue(success.get() > 0);
+        assertEquals(0, error.get());
+    }
```

Comment:
FWIW this rephrasing of the method fails for me at the same proportion (~95%) with the fix rolled back, and it allows removal of half the tests because we're combining assertions for cloned and non-cloned cases. 

```suggestion
    /**
     * Verifies that a read-only struct's large nested struct can be accessed via multiple threads concurrently.
     * @param makeParentReadOnly if true, the top-level struct will be made only (cascading to all children); if false,
     *                           only the child struct to be accessed concurrently will be made read-only.
     */
    public void testReadOnlyIonStructMultithreadedNestedAccess(boolean makeParentReadOnly) {
        IonStruct struct = createIonStructWithLargeChildStructs();
        IonStruct clone = createIonStructWithLargeChildStructs().clone();

        if (makeParentReadOnly) {
            struct.makeReadOnly();
            clone.makeReadOnly();
        } else {
            struct.get("child2").makeReadOnly();
            clone.get("child2").makeReadOnly();
        }

        Consumer<IonStruct> ionStructConsumer = struct1 -> {
                IonStruct child2 = (IonStruct) struct1.get("child2");
                String value = child2 == null || child2.get("target") == null
                        ? null
                        : ((IonStruct) child2.get("target")).get("a").toString();

                assertNotNull(value);
        };

        IntStream.range(0, 100)
                .mapToObj($ -> CompletableFuture.runAsync(() -> {
                    ionStructConsumer.accept(struct);
                    ionStructConsumer.accept(clone);
                }))
                .collect(Collectors.toList()) // We have to collect to force failure here
                .forEach(CompletableFuture::join);
    }
```